router-sink/src/main/java/org/springframework/cloud/stream/module/router/sink/RouterSink.java

```diff
@@ -0,0 +1,95 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.router.sink;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.binding.BinderAwareChannelResolver;
+import org.springframework.cloud.stream.config.SpelExpressionConverterConfiguration;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.cloud.stream.module.common.ScriptVariableGeneratorConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.groovy.GroovyScriptExecutingMessageProcessor;
+import org.springframework.integration.router.AbstractMappingMessageRouter;
+import org.springframework.integration.router.ExpressionEvaluatingRouter;
+import org.springframework.integration.router.MethodInvokingRouter;
+import org.springframework.integration.scripting.RefreshableResourceScriptSource;
+import org.springframework.integration.scripting.ScriptVariableGenerator;
+import org.springframework.scripting.ScriptSource;
+
+/**
+ * A sink module that routes to one or more named channels.
+ *
+ * @author Gary Russell
+ */
+@EnableBinding(Sink.class)
+@Import({ SpelExpressionConverterConfiguration.class, ScriptVariableGeneratorConfiguration.class })
+@EnableConfigurationProperties(RouterSinkProperties.class)
+public class RouterSink {
+
+	@Autowired
+	private RouterSinkProperties properties;
+
+	@Autowired
+	private ScriptVariableGenerator scriptVariableGenerator;
+
+	@Autowired
+	@Bindings(RouterSink.class)
+	Sink channels;
+
+	@Bean
+	@ServiceActivator(inputChannel=Sink.INPUT)
+	public AbstractMappingMessageRouter router(BinderAwareChannelResolver channelResolver,
+			ConfigurableBeanFactory beanFactory) throws Exception {
+		AbstractMappingMessageRouter router;
+		if (this.properties.getScript() != null) {
+			router = new MethodInvokingRouter(scriptProcessor(beanFactory));
+		}
+		else {
+			router = new ExpressionEvaluatingRouter(this.properties.getExpression());
+		}
+		router.setDefaultOutputChannelName(this.properties.getDefaultOutputChannel());
+		router.setResolutionRequired(this.properties.isResolutionRequired());
+		Map<String, String> mappings = new LinkedHashMap<String, String>();
+		int n = 0;
+		for (String value : this.properties.getValues()) {
+			mappings.put(value, this.properties.getDestinations()[n++]);
+		}
```

Comment:
Can't we use the similar `Properties` approach for the `ChannelMappings` as well?
Instead of this strange `values/destinations` pair.
Especially when we have already `MappingMessageRouterManagement.replaceChannelMappings()`